src/amp-story-embed.js

```diff
@@ -109,19 +113,51 @@ export class AmpStoryEmbed {
 
     this.iframeEl_.onload = () => {
       this.rootEl_.classList.remove(LoadStateClass.LOADING);
+      this.element_.classList.remove(LoadStateClass.LOADING);
       this.rootEl_.classList.add(LoadStateClass.LOADED);
+      this.element_.classList.add(LoadStateClass.LOADED);
     };
     this.iframeEl_.onerror = () => {
       this.rootEl_.classList.remove(LoadStateClass.LOADING);
+      this.element_.classList.remove(LoadStateClass.LOADING);
       this.rootEl_.classList.add(LoadStateClass.ERROR);
+      this.element_.classList.add(LoadStateClass.ERROR);
     };
   }
-}
 
-const doc = self.document;
-const embeds = doc.getElementsByTagName('amp-story-embed');
-for (let i = 0; i < embeds.length; i++) {
-  const embed = embeds[i];
-  const embedImpl = new AmpStoryEmbed(doc, embed);
-  embedImpl.build();
+  /**
+   * @return {!Promise}
+   * @public
+   */
+  layoutCallback() {
+    // TODO: Layout all child iframes.
+    return this.layoutIframe_(0);
+  }
+
+  /**
+   * @param {number} index
+   * @return {!Promise}
+   * @private
+   */
+  layoutIframe_(index) {
+    const story = this.stories_[index];
+    this.iframeEl_.setAttribute('src', story.href);
+    return Promise.resolve();
+  }
 }
+
+self.onload = () => {
+  if (isAMP) {
```

Comment:
I think we should just remove this entirely, and assume that this script will strictly be imported in non-AMP environments.  It's not valid AMP anyway.

Reply:
SG. Removed it.